pkg/cluster/check/etcd.go

```diff
@@ -0,0 +1,63 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package check
+
+import (
+	"context"
+	"fmt"
+
+	machineapi "github.com/talos-systems/talos/pkg/machinery/api/machine"
+)
+
+// EtcdConsistentAssertion checks that etcd membership is consistent across nodes.
+//nolint:gocyclo
+func EtcdConsistentAssertion(ctx context.Context, cluster ClusterInfo) error {
+	cli, err := cluster.Client()
+	if err != nil {
+		return err
+	}
+
+	resp, err := cli.EtcdMemberList(ctx, &machineapi.EtcdMemberListRequest{})
```

Comment:
this way we're talking to a random node in the cluster.

what we really want to do is to talk to all control plane nodes: `client.WithNodes(ctx, cluster.NodesByType(machine.TypeInit, machine.TypeControlPlane)...)`

this way we get responses from all control plane nodes

then each response line is a single member coming from one of the nodes, node info which delivered the response will be in message.Metadata, we should build a list for each node and they should match

Reply:
this code might help: https://github.com/siderolabs/talos/blob/5ec4e9096e98c8bf5e27b1a78171498dbc94b23c/cmd/talosctl/cmd/talos/etcd.go#L95-L129

it's more complicated than we need (it supports talking without node context, we don't need it here, in our case member.Hostname should be always set to the node we were talking to)

Reply:
and yep, we should get a message back from each node we requested with `client.WithNodes`